shacl12-ui/index.html

```diff
@@ -801,8 +801,592 @@ <h3>Document Conventions</h3>
+            a drop-down editor for all instances of the target class (based on <code>sh:class</code> of the property).
+            Typically only used for classes that have few instances.
+          </p>
+          <pre class="example-shapes">
+ex:Person-homeCountry
+	a sh:PropertyShape ;
+	sh:path ex:homeCountry ;
+	<b>sh:class ex:Country ;
+	shui:editor shui:InstancesSelectEditor ;</b>
+	...</pre>
+        </section>
+        <section id="RichTextEditor">
+          <h4>shui:RichTextEditor</h4>
+          <p>
+            <b>Score:</b>
+            10 for <code>rdf:HTML</code> literals.
+            0 otherwise.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            a rich text editor to enter the lexical value of a literal and a drop-down to select language.
+            The selected language is stored in the HTML <code>lang</code> attribute of the root node in the HTML DOM tree.
+          </p>
+          <img src="images/editors/RichTextEditor.png" />
+          <pre class="example-shapes">
+ex:Concept-definition
+	a sh:PropertyShape ;
+	sh:path skos:definition ;
+	<b>sh:datatype rdf:HTML ;</b>
+	...</pre>
+        </section>
+        <section id="SubClassEditor">
+          <h4>shui:SubClassEditor</h4>
+          <p>
+            <b>Score:</b>
+            <code>null</code> i.e. this should be selected explicitly through a <code>shui:editor</code> statement.
+            However, this widget is typically only used if the property has a
+            <a href="https://datashapes.org/constraints.html#RootClassConstraintComponent"><code>dash:rootClass</code></a>
+            constraint, or (at minimum) only allows classes as values.
+          </p>
+          <p>
+            <b>Rendering:</b>
+            This may be an auto-complete widget to select a class or a class hierarchy widget, or a combination thereof.
+            The permissible values are a given class or its subclasses, defaulting to <code>rdfs:Resource</code>.
+            This is typically used with <code>dash:rootClass</code> to allow the user to select a subclass of the given root class.
+          </p>
+          <pre class="example-shapes">
+ex:Drug-impactedCell
+	a sh:PropertyShape ;
+	sh:path ex:impactedCell ;
+	<b>dash:rootClass obo:CL_0000000 ;
+	shui:editor shui:SubClassEditor ;</b>
+	...</pre>
+        </section>
+        <section id="TextAreaEditor">
+          <h4>shui:TextAreaEditor</h4>
+          <p>
+            <b>Score:</b>
+            0 if the property is marked <code>sh:singleLine true</code>.
+            20 if the value is an <code>xsd:string</code> literal and <code>sh:singleLine false</code>.
+            5 if the value is an <code>xsd:string</code> literal.
+            2 if the property has <code>xsd:string</code> among the permissible datatypes.
+            <code>null</code> if the property has a custom datatype (not from xsd or rdf namespaces but for example <code>geo:wktLiteral</code>).
+            0 otherwise.
```

Comment:
Such long lists may be better *as* lists (I recommend `<ul>` to prevent confusion between a numbered item in an `<ol>` and the score discussed in that item. I think `<ol>` is fine, since these are all disjoint, and *can* be presented in any order.), as the clarity of this manually linebroken source does not persist when all these lines fold into a single wrapping line. Such `<ul>` or `<ol>` lists should probably be used throughout, whenever there are multiple `Score:` lines.
```suggestion
            0 if the property is marked <code>sh:singleLine true</code>.
            20 if the value is an <code>xsd:string</code> literal and <code>sh:singleLine false</code>.
            5 if the value is an <code>xsd:string</code> literal.
            2 if the property has <code>xsd:string</code> among the permissible datatypes.
            <code>null</code> if the property has a custom datatype (not from <code>xsd:</code> nor <code>rdf:</code> namespaces, but, for example, <code>geo:wktLiteral</code>).
            0 otherwise.
```